Hearthstone Deck Tracker/Windows/DeckScreenshotWindow.xaml.cs

```diff
@@ -0,0 +1,187 @@
+
+		public double LabelsHeight => DeckTitleContainer.ActualHeight + DeckFormatPanel.ActualHeight
+			+ SetDustPanel.ActualHeight + BrandContainer.ActualHeight + 42;
+
+		public void UpdateWindowHeight()
+		{
+			this.Height = StackPanelMain.ActualHeight + 42;
+		}
+
+		private ImageBrush DeckHeaderBackground(string deckClass)
+		{
+			var heroId = ClassToID(deckClass);
+			DrawingGroup DrawingGroup = new DrawingGroup();
+			DrawingGroup.Children.Add(new ImageDrawing(new BitmapImage(new Uri(
+				$"Images/Bars/{heroId}.png", UriKind.Relative)), new Rect(54, 0, 130, 34)));
+			DrawingGroup.Children.Add(new ImageDrawing(new BitmapImage(new Uri(
+				"Images/Themes/Bars/dark/fade.png", UriKind.Relative)), new Rect(0, 0, 183, 34)));
+
+			return new ImageBrush {
+				ImageSource = new DrawingImage(DrawingGroup),
+				AlignmentX = AlignmentX.Left,
+				Stretch = Stretch.UniformToFill
+			};
+		}
+
+		private string GetTagText(Deck deck)
+		{
+			List<string> predefined = new List<string>() {
+				"Midrange",
+				"Aggro",
+				"Control",
+				"Tempo",
+				"Combo"
+			};
+			if(deck.Tags.Count > 0)
+			{
+				foreach(var tag in predefined)
+				{
+					var found = deck.Tags.Find(
+						f => f.ToLowerInvariant().Replace("-", "") == tag.ToLowerInvariant());
+					if(found != null)
+						return found;
+				}
+			}
+			return deck.Class;
+		}
+
+		private string GetFormatText(Deck deck)
+		{
+			var allTags = deck.TagList.ToLowerInvariant();
+			if(deck.IsArenaDeck)
+				return "Arena";
+			else if(allTags.Contains("brawl"))
+				return "Brawl";
+			else if(allTags.Contains("adventure") || allTags.Contains("pve"))
+				return "Adventure";
+			else if(deck.StandardViable)
+				return "Standard";
+			else
+				return "Wild";
+		}
+
+		private void ShowFormatIcon(Deck deck)
+		{
+			RectIconStandard.Visibility = Collapsed;
+			RectIconWild.Visibility = Collapsed;
+			RectIconArena.Visibility = Collapsed;
+			RectIconBrawl.Visibility = Collapsed;
+			RectIconAdventure.Visibility = Collapsed;
+
+			var allTags = deck.TagList.ToLowerInvariant();
+			if(deck.IsArenaDeck)
+				RectIconArena.Visibility = Visible;
+			else if(allTags.Contains("brawl"))
+				RectIconBrawl.Visibility = Visible;
+			else if(allTags.Contains("adventure") || allTags.Contains("pve"))
+				RectIconAdventure.Visibility = Visible;
+			else if(deck.StandardViable)
+				RectIconStandard.Visibility = Visible;
+			else
+				RectIconWild.Visibility = Visible;
+		}
+
+		private int TotalDust(Deck deck)
+		{
+			List<string> nonCraftableSets = new List<string>() {
+				"Curse of Naxxramas",
+				"Blackrock Mountain",
+				"League of Explorers",
+				"Basic"
+			};
+			List<string> nonCraftableCards = new List<string>() {
+				"OG_280", "OG_281"
+			};
+
+			return deck.Cards
+				.Where(c => !nonCraftableSets.Contains(c.Set) && !nonCraftableCards.Contains(c.Id))
+				.Sum(c => c.DustCost * c.Count);
+		}
+
+		private string ClassToID(string klass)
+		{
+			switch(klass.ToLowerInvariant())
+			{

```

Comment:
Should use `HearthDb.CardIds` instead of raw card id strings. (E.g. `Collectible.Mage.JainaProudmoore`)